npm/debugTools/networkTupleProcessor/tupleProcessor.go

```diff
@@ -0,0 +1,367 @@
+package processor
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os/exec"
+	"strconv"
+	"strings"
+
+	"github.com/Azure/azure-container-networking/npm"
+	converter "github.com/Azure/azure-container-networking/npm/debugTools/dataplaneConverter"
+	"github.com/Azure/azure-container-networking/npm/debugTools/pb"
+	"github.com/Azure/azure-container-networking/npm/util"
+	"google.golang.org/protobuf/encoding/protojson"
+)
+
+type Processor struct {
+}
+
+type Tuple struct {
+	RuleType  string `json:"ruleType"`
+	Direction string `json:"direction"`
+	SrcIP     string `json:"srcIP"`
+	SrcPort   string `json:"srcPort"`
+	DstIP     string `json:"dstIP"`
+	DstPort   string `json:"dstPort"`
+	Protocol  string `json:"protocol"`
+}
+
+type Input struct {
+	Content string
+	Type    InputType
+}
+
+type InputType int32
+
+const (
+	IPADDRS  InputType = 0
+	PODNAME  InputType = 1
+	INTERNET InputType = 2
+)
+
+// GetNetworkTuple returns a list of hit rules between the source and the destination in JSON format and a list of tuples from those rules. Filenames following the format, cacheFile first and then iptable-save file
+// optional for debugging
+func (p *Processor) GetNetworkTuple(src, dst *Input, filenames ...string) ([][]byte, []*Tuple) {
+	c := &converter.Converter{}
+	var (
+		cacheObj      *converter.NPMCache
+		allRules      []*pb.RuleResponse
+		srcPod        *npm.NpmPod
+		dstPod        *npm.NpmPod
+		iptableBuffer = bytes.NewBuffer(nil)
+		tableName     = "filter"
+	)
+
+	// hacky way to make it works for testing
+	if len(filenames) > 1 {
+		cacheObj = c.GetNpmCache(filenames[0])
```

Comment:
it is very easy for us to rearrange that order in the caller function, so lets have specific filenames

Reply:
This GetNPmCahce might a duplicate  call, since below in GetProtobufRulesFromIptable -> inti container also we are getting it